nearform/optic-expo
# Feat/notification permission

closes #1265 

Changes made :
- Changed around the order of when "setNotificationChannelAsync()' is called as that is required on Android before requesting permissions( for the native popup to come up).
- reworked token hook so that we don't have to request notification permissions when we 
- Added a new popup modal that links to the settings app when permissions are not accepted. 

## Setup Instructions for testing
### For Expo Go:

*On your phone*
- If we already have the app installed we need to clear data and reinstall the expo go app, to get rid of the permissions token.
- In the expo app Sign in with an account that is in the nearform org ( ask @Simone if you are not already in the org)

- Scan the QR code from the repo, whould be the first comment on this PR. This should work for both iPhone and android if you scan the correct codes.

### Native:

For native we should clone the repo and refer to the setup guide here:

https://github.com/nearform/optic-expo
- Once you have installed the dependencies using 'yarn'. look at the relevant section for your device. 


### Andriod

- For android, we need to make sure that a real Android device is plugged in. (*note* USB debugging should be enabled on your device, this is usually in settings > developer options), also if you are on Samsung you should disable/ uninstall secure folder before trying to build on your phone, I was only able to get this to work after I did that.

- We need to make sure we have Andriod Studio installed on your computer(macOS) and configured as the guide says here: 
https://docs.expo.dev/workflow/android-studio-emulator/

- We need to also have OpenJDK 11 installed which can be found here: https://formulae.brew.sh/formula/openjdk@11

- After the installation, the output should give instructions on the relevant PATH variables and what to do with them.

- Once this is done we simply run `yarn android` in the terminal of the project with an Android device connected to your computer via USB. 

### iOS
- We need to Xcode installed from the app store on our computer(macOS).
- Install expo go:(https://apps.apple.com/us/app/expo-go/id982107779) on your device.
- Confirm your device is in [developer mode](https://developer.apple.com/documentation/xcode/enabling-developer-mode-on-a-device)
- Connect the iOS device to your Mac and select - trust computer when prompted.
- Setup code signing certificates in Xcode for development (https://github.com/expo/fyi/blob/main/setup-xcode-signing.md)
- In Xcode select your device as deployment target and select the development certificate (you may need to connect the device to your account if it was not done before.
![select device](https://user-images.githubusercontent.com/84892/191241139-5db5b905-228f-4af2-a633-47f601de993a.png)
- run `expo run:ios --device`

# Testing instructions
- Testing notifications
Once we open the app, if there is no previous data, we should get a popup asking to enable push notification permissions. 
We need to confirm the respective native popup appears for both Android & Ios. This will only appear once, If the notifications are declined on the first go, clicking enable should navigate us to the app settings page so we can manually enable notifications. 
- Test that tokens are added after accepting notifications.
- If the notifications are accepted, we should be able to add a secret and a token, click the "+" Icon on the bottom right of the screen and add a secret. 
- Click 'add token' and then create a token, you should be able to successfully create a token without errors.
